unstructured_inference/models/unstructuredmodel.py

```diff
@@ -55,6 +63,116 @@ def __call__(self, x: Image) -> List[LayoutElement]:
         """Inference using function call interface."""
         return super().__call__(x)
 
+    @staticmethod
+    def enhance_regions(
+        elements: List[LayoutElement],
+        iom_to_merge: float = 0.3,
+    ) -> List[LayoutElement]:
+        """This function traverses all the elements and either deletes nested elements,
+        or merges or splits them depending on the iom score for both regions"""
+        intersections_mtx = intersections(*elements)
+
+        for i, row in enumerate(intersections_mtx):
+            first = elements[i]
+            if first:
+                # We get only the elements which intersected
+                indices_to_check = np.where(row)[0]
+                # Delete the first element, since it will always intersect with itself
+                indices_to_check = indices_to_check[indices_to_check != i]
+                if len(indices_to_check) == 0:
+                    continue
+                if len(indices_to_check) > 1:  # sort by iom
```

Comment:
I don't think we need this check since we have the two lines above:? Then we can reduce indentation of the block below

Reply:
You mean `len(index_to_check)>1`, right?
That is needed because will perform unnecessary processing when just have one element